### DISCUSSION TITLE
Unable to change cache driver when using RateLimiter class

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#43287] [Ideas]
When using RateLimiter class like so:

```php
use Illuminate\Support\Facades\RateLimiter;

RateLimiter::attempt(...)
```

You can not use another driver at the moment, it only uses the default cache driver.

For example my default cache driver is database, But i wanna use the `file` driver for that expression

**Comment by BrekiTomasson** [comment]
`RateLimiter` always uses the `cache` driver since it's meant to be the fastest lookup you have available, but I agree that it could be interesting to have the option of doing something like `RateLimiter::using('file')->attempt(...)`.

That said, my advice would be to install `redis` and use it as your `cache`, `queue` and `session` drivers.
